manifests/base.pp

```diff
@@ -43,6 +44,17 @@
     }
   }
 
+  if $rsyslog::switch_default_syslog {
```

Comment:
instead of this param which calls an exec that only makes sense on AIX, suggest conditional logic based off of the platform such as

```puppet
if $facts['os']['name'] == 'AIX' {
```

Reply:
If I change to your suggestion it would mean that the following command would run every puppet run trying to do a idempotent check: `odmget -q \"subsysname = 'syslogd'\" SRCsubsys | grep rsyslog`.  This is not a desirable behavior. 

Additionally, some users want this module to install rsyslog but also want to control when syslog is changed over to rsyslog.  AFAIK, rsyslog is not the default for AIX.

I did update to use `if $rsyslog::switch_default_syslog and $::facts['os']['name'] == 'AIX' {` to prevent any accidental usage from non AIX systems.